parsec/core/gui/files_widget.py

```diff
@@ -328,24 +320,55 @@ def on_paste_clicked(self):
             while str(base_name) != base_name.stem:
                 base_name = pathlib.Path(base_name.stem)
             count = 2
+            timestamp_already_added = False
             base_name = str(base_name)
             while True:
                 try:
                     dst = self.current_directory / file_name
-                    if self.clipboard.status == Clipboard.Status.Cut:
-                        self.jobs_ctx.run(self.workspace_fs.move, src, dst)
+                    source_workspace = (
+                        self.clipboard.workspace if self.clipboard.workspace is not self else None
+                    )
+                    if self.clipboard.status == self.clipboard.Status.Cut:
+                        self.jobs_ctx.run(
```

Comment:
Isn't it time to move this while loop to a single job instead of looping over repeated jobs?

Reply:
I don't know, I just adapted the existing code to enable that kind of copy, I don't know PyQt best practices, this is Max's code I guess... @Max-7 wdyt?